File: BackEnd/Application/UseCases/UserGroups/UpdateUserGroup/UpdateUserGroupCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;
using Microsoft.EntityFrameworkCore;

namespace Application.UseCases.UserGroups.UpdateUserGroup;

public class UpdateUserGroupCommandHandler(IRepository<UserGroup, Guid> userGroupRepository, IUnitOfWork unitOfWork, IRepository<User, Guid> userRepository) : IRequestHandler<UpdateUserGroupCommand, bool>
{
    public async Task<bool> Handle(UpdateUserGroupCommand request, CancellationToken cancellationToken)
    {
        var users = await userRepository.Where(x => request.UserIds.Contains(x.Id)).ToListAsync(cancellationToken);
        var userGroup = await userGroupRepository.Where(x => x.Id.Equals(request.Id)).Include(x => x.Users).FirstOrDefaultAsync(cancellationToken);
        if (userGroup == null) return false;
        
        userGroup.Update(request.Name, users);
        await userGroupRepository.UpdateAsync(userGroup);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/user_groups/update_user_group/update_user_group_command_handler.dart
import 'package:enterprise_management/application/use_cases/user_groups/update_user_group/update_user_group_command.dart';
import 'package:enterprise_management/infrastructure/repositories/user_group_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateUserGroupCommandHandler extends ICommandHandler<UpdateUserGroupCommand, bool> {
  const UpdateUserGroupCommandHandler({required this._userGroupRepository});
  final IUserGroupRepository _userGroupRepository;

  @override
  Future<Result<bool>> handle(UpdateUserGroupCommand command) async {
    final res = await _userGroupRepository.updateUserGroupById(id: command.id, name: command.name, userIds: command.userIds);
    return Result.success(res);
  }

}